README.md

```diff
@@ -28,6 +28,17 @@ function f(dx,x)
   dx[end] = x[end-1] - 2x[end]
   nothing
 end
+
+function g(x) # out-of-place
+  global fcalls += 1
+  dx = zero(x)
```

Comment:
we should make this test be on static arrays instead, so it tests that no mutation is used by failing on mutation